### ISSUE TITLE
Bug: NPE when INSERT INTO b SELECT * FROM a

### ISSUE THREAD

**Issue by proddata** [crate/crate#10874]
Maybe directly related to #10851

---

**CrateDB version**:
4.3.2

**Environment description**:
CrateDB Cloud - Dev Cluster
- 3x 0.25 vCPUs
- 3x 2 GB Memory (0.5 GB Heap)

**Problem description**:

- 1 TIMESTAMP column (PK)
- ~ 190 DOUBLE PRECISION columns
- ~ 10 TEXT columns

```sql
CREATE TABLE open_pv (
    DateTime TIMESTAMPTZ PRIMARY KEY ,
  temperature_outdoor_fantasytown_weather DOUBLE PRECISION,
  ...
  c_phase_voltage_brightcounty_pv DOUBLE PRECISION);
```

- 103.2 MB (Size primary shard)
- 23,422 records

when creating a second table `open_pv2` with the schema as `open_pv` and trying to copy the data with

```sql
INSERT INTO open_pv2 SELECT * FROM open_pv
-- INSERT ERROR (0.000 sec)
```
this leads to the following error
```java
NullPointerException[Cannot invoke "org.elasticsearch.common.io.stream.Writeable$Reader.read(org.elasticsearch.common.io.stream.StreamInput)" because "this.reader" is null]
```

---

Limiting the number by using `LIMIT 10000;` the query succedes
```sql
INSERT INTO open_pv2 SELECT * FROM open_pv LIMIT 10000;
-- INSERT OK, 10000 rows affected (37.742 sec)
```

**Steps to reproduce**:

1. Create Table `open_pv`
```sql
CREATE TABLE open_pv (
    DateTime TIMESTAMPTZ PRIMARY KEY ,
  temperature_outdoor_fantasytown_weather DOUBLE PRECISION,
  ...
  c_phase_voltage_brightcounty_pv DOUBLE PRECISION);
```
2. Insert ~25.000 rows

3. Create Table `open_pv2`
```sql
CREATE TABLE open_pv2 (
    DateTime TIMESTAMPTZ PRIMARY KEY ,
  temperature_outdoor_fantasytown_weather DOUBLE PRECISION,
  ...
  c_phase_voltage_brightcounty_pv DOUBLE PRECISION);
```

4. Try to copy table with `INSERT SELECT FROM`
```sql
INSERT INTO open_pv2 SELECT * FROM open_pv
-- INSERT ERROR (0.000 sec)
```

**Comment by proddata** [comment]
Just tested the same on a bigger cluster -> same result

adding (even very high) LIMITs mitigates the issue ... 

i.e.
```SQL
INSERT INTO open_pv2 SELECT * from open_pv LIMIT 1000000;
```
works

**Comment by seut** [comment]
@proddata Thank you for reporting this. Unfortunately we have problems reproducing this.
Can you reproduce the issue even with the stripped table schema you posted? So following exactly the described steps with only 3 columns inside each table?

**Comment by proddata** [comment]
complete schema
```sql
CREATE TABLE open_pv (
    DateTime TIMESTAMPTZ PRIMARY KEY ,
temperature_outdoor_fantasytown_weather DOUBLE PRECISION,
temperature_outdoor_sunnycity_weather DOUBLE PRECISION,
temperature_outdoor_happyville_weather DOUBLE PRECISION,
temperature_outdoor_eastwood_weather DOUBLE PRECISION,
temperature_outdoor_brightcounty_weather DOUBLE PRECISION,
humidity_outdoor_fantasytown_weather DOUBLE PRECISION,
humidity_outdoor_sunnycity_weather DOUBLE PRECISION,
humidity_outdoor_happyville_weather DOUBLE PRECISION,
humidity_outdoor_eastwood_weather DOUBLE PRECISION,
humidity_outdoor_brightcounty_weather DOUBLE PRECISION,
dew_point_outdoor_fantasytown_weather DOUBLE PRECISION,
dew_point_outdoor_sunnycity_weather DOUBLE PRECISION,
dew_point_outdoor_happyville_weather DOUBLE PRECISION,
dew_point_outdoor_eastwood_weather DOUBLE PRECISION,
dew_point_outdoor_brightcounty_weather DOUBLE PRECISION,
wind_speed_fantasytown_weather DOUBLE PRECISION,
wind_speed_sunnycity_weather DOUBLE PRECISION,
wind_speed_happyville_weather DOUBLE PRECISION,
wind_speed_eastwood_weather DOUBLE PRECISION,
wind_speed_brightcounty_weather DOUBLE PRECISION,
peak_wind_speed_fantasytown_weather DOUBLE PRECISION,
peak_wind_speed_sunnycity_weather DOUBLE PRECISION,
peak_wind_speed_happyville_weather DOUBLE PRECISION,
peak_wind_speed_eastwood_weather DOUBLE PRECISION,
peak_wind_speed_brightcounty_weather DOUBLE PRECISION,
wind_chill_fantasytown_weather DOUBLE PRECISION,
wind_chill_sunnycity_weather DOUBLE PRECISION,
wind_chill_happyville_weather DOUBLE PRECISION,
wind_chill_eastwood_weather DOUBLE PRECISION,
wind_chill_brightcounty_weather DOUBLE PRECISION,
perceived_heat_outdoor_fantasytown_weather DOUBLE PRECISION,
perceived_heat_outdoor_sunnycity_weather DOUBLE PRECISION,
perceived_heat_outdoor_happyville_weather DOUBLE PRECISION,
perceived_heat_outdoor_eastwood_weather DOUBLE PRECISION,
perceived_heat_outdoor_brightcounty_weather DOUBLE PRECISION,
pressure_fantasytown_weather DOUBLE PRECISION,
pressure_sunnycity_weather DOUBLE PRECISION,
pressure_happyville_weather DOUBLE PRECISION,
pressure_eastwood_weather DOUBLE PRECISION,
pressure_brightcounty_weather DOUBLE PRECISION,
rainfall_fantasytown_weather DOUBLE PRECISION,
rainfall_sunnycity_weather DOUBLE PRECISION,
rainfall_happyville_weather DOUBLE PRECISION,
rainfall_eastwood_weather DOUBLE PRECISION,
rainfall_brightcounty_weather DOUBLE PRECISION,
rainfall_rate_fantasytown_weather DOUBLE PRECISION,
rainfall_rate_sunnycity_weather DOUBLE PRECISION,
rainfall_rate_happyville_weather DOUBLE PRECISION,
rainfall_rate_eastwood_weather DOUBLE PRECISION,
rainfall_rate_brightcounty_weather DOUBLE PRECISION,
solar_radiation_fantasytown_weather DOUBLE PRECISION,
solar_radiation_sunnycity_weather DOUBLE PRECISION,
solar_radiation_happyville_weather DOUBLE PRECISION,
solar_radiation_eastwood_weather DOUBLE PRECISION,
solar_radiation_brightcounty_weather DOUBLE PRECISION,
peak_solar_radiation_radiation_fantasytown_weather DOUBLE PRECISION,
peak_solar_radiation_radiation_sunnycity_weather DOUBLE PRECISION,
peak_solar_radiation_radiation_happyville_weather DOUBLE PRECISION,
peak_solar_radiation_radiation_eastwood_weather DOUBLE PRECISION,
peak_solar_radiation_radiation_brightcounty_weather DOUBLE PRECISION,
temperature_indoor_fantasytown_weather DOUBLE PRECISION,
temperature_indoor_sunnycity_weather DOUBLE PRECISION,
temperature_indoor_happyville_weather DOUBLE PRECISION,
temperature_indoor_eastwood_weather DOUBLE PRECISION,
temperature_indoor_brightcounty_weather DOUBLE PRECISION,
humidity_indoor_fantasytown_weather DOUBLE PRECISION,
humidity_indoor_sunnycity_weather DOUBLE PRECISION,
humidity_indoor_happyville_weather DOUBLE PRECISION,
humidity_indoor_eastwood_weather DOUBLE PRECISION,
humidity_indoor_brightcounty_weather DOUBLE PRECISION,
dew_point_indoor_fantasytown_weather DOUBLE PRECISION,
dew_point_indoor_sunnycity_weather DOUBLE PRECISION,
dew_point_indoor_happyville_weather DOUBLE PRECISION,
dew_point_indoor_eastwood_weather DOUBLE PRECISION,
dew_point_indoor_brightcounty_weather DOUBLE PRECISION,
perceived_heat_indoor_fantasytown_weather DOUBLE PRECISION,
perceived_heat_indoor_sunnycity_weather DOUBLE PRECISION,
perceived_heat_indoor_happyville_weather DOUBLE PRECISION,
perceived_heat_indoor_eastwood_weather DOUBLE PRECISION,
perceived_heat_indoor_brightcounty_weather DOUBLE PRECISION,
air_density_indoor_fantasytown_weather DOUBLE PRECISION,
air_density_indoor_sunnycity_weather DOUBLE PRECISION,
air_density_indoor_happyville_weather DOUBLE PRECISION,
air_density_indoor_eastwood_weather DOUBLE PRECISION,
air_density_indoor_brightcounty_weather DOUBLE PRECISION,
evapotranspiration_fantasytown_weather DOUBLE PRECISION,
evapotranspiration_sunnycity_weather DOUBLE PRECISION,
evapotranspiration_happyville_weather DOUBLE PRECISION,
evapotranspiration_eastwood_weather DOUBLE PRECISION,
evapotranspiration_brightcounty_weather DOUBLE PRECISION,
wind_direction_fantasytown_weather TEXT,
wind_direction_sunnycity_weather TEXT,
wind_direction_happyville_weather TEXT,
wind_direction_eastwood_weather TEXT,
wind_direction_brightcounty_weather TEXT,
peak_wind_direction_fantasytown_weather TEXT,
peak_wind_direction_sunnycity_weather TEXT,
peak_wind_direction_happyville_weather TEXT,
peak_wind_direction_eastwood_weather TEXT,
peak_wind_direction_brightcounty_weather TEXT,
single_phase_voltage_cloudington_pv DOUBLE PRECISION,
single_phase_voltage_flowervillage_pv DOUBLE PRECISION,
single_phase_voltage_happyville_pv DOUBLE PRECISION,
single_phase_voltage_eastwood_pv DOUBLE PRECISION,
single_phase_voltage_sunrisebay_pv DOUBLE PRECISION,
single_phase_voltage_brightcounty_pv DOUBLE PRECISION,
generator_current_cloudington_pv DOUBLE PRECISION,
generator_current_flowervillage_pv DOUBLE PRECISION,
generator_current_happyville_pv DOUBLE PRECISION,
generator_current_eastwood_pv DOUBLE PRECISION,
generator_current_sunrisebay_pv DOUBLE PRECISION,
generator_current_brightcounty_pv DOUBLE PRECISION,
imported_current_cloudington_pv DOUBLE PRECISION,
imported_current_flowervillage_pv DOUBLE PRECISION,
imported_current_happyville_pv DOUBLE PRECISION,
imported_current_eastwood_pv DOUBLE PRECISION,
imported_current_sunrisebay_pv DOUBLE PRECISION,
imported_current_brightcounty_pv DOUBLE PRECISION,
power_generation_cloudington_pv DOUBLE PRECISION,
power_generation_flowervillage_pv DOUBLE PRECISION,
power_generation_happyville_pv DOUBLE PRECISION,
power_generation_eastwood_pv DOUBLE PRECISION,
power_generation_sunrisebay_pv DOUBLE PRECISION,
power_generation_brightcounty_pv DOUBLE PRECISION,
imported_power_cloudington_pv DOUBLE PRECISION,
imported_power_flowervillage_pv DOUBLE PRECISION,
imported_power_happyville_pv DOUBLE PRECISION,
imported_power_eastwood_pv DOUBLE PRECISION,
imported_power_sunrisebay_pv DOUBLE PRECISION,
imported_power_brightcounty_pv DOUBLE PRECISION,
reactive_power_cloudington_pv DOUBLE PRECISION,
reactive_power_flowervillage_pv DOUBLE PRECISION,
reactive_power_happyville_pv DOUBLE PRECISION,
reactive_power_eastwood_pv DOUBLE PRECISION,
reactive_power_sunrisebay_pv DOUBLE PRECISION,
reactive_power_brightcounty_pv DOUBLE PRECISION,
imported_reactive_power_cloudington_pv DOUBLE PRECISION,
imported_reactive_power_flowervillage_pv DOUBLE PRECISION,
imported_reactive_power_happyville_pv DOUBLE PRECISION,
imported_reactive_power_eastwood_pv DOUBLE PRECISION,
imported_reactive_power_sunrisebay_pv DOUBLE PRECISION,
imported_reactive_power_brightcounty_pv DOUBLE PRECISION,
apparent_power_cloudington_pv DOUBLE PRECISION,
apparent_power_flowervillage_pv DOUBLE PRECISION,
apparent_power_happyville_pv DOUBLE PRECISION,
apparent_power_eastwood_pv DOUBLE PRECISION,
apparent_power_sunrisebay_pv DOUBLE PRECISION,
apparent_power_brightcounty_pv DOUBLE PRECISION,
imported_apparent_power_cloudington_pv DOUBLE PRECISION,
imported_apparent_power_flowervillage_pv DOUBLE PRECISION,
imported_apparent_power_happyville_pv DOUBLE PRECISION,
imported_apparent_power_eastwood_pv DOUBLE PRECISION,
imported_apparent_power_sunrisebay_pv DOUBLE PRECISION,
imported_apparent_power_brightcounty_pv DOUBLE PRECISION,
voltage_harmonic_distortion_cloudington_pv DOUBLE PRECISION,
voltage_harmonic_distortion_flowervillage_pv DOUBLE PRECISION,
voltage_harmonic_distortion_happyville_pv DOUBLE PRECISION,
voltage_harmonic_distortion_eastwood_pv DOUBLE PRECISION,
voltage_harmonic_distortion_sunrisebay_pv DOUBLE PRECISION,
voltage_harmonic_distortion_brightcounty_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_cloudington_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_flowervillage_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_happyville_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_eastwood_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_sunrisebay_pv DOUBLE PRECISION,
generated_current_harmonic_distortion_brightcounty_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_cloudington_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_flowervillage_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_happyville_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_eastwood_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_sunrisebay_pv DOUBLE PRECISION,
imported_current_harmonic_distortion_brightcounty_pv DOUBLE PRECISION,
phase_frequency_cloudington_pv DOUBLE PRECISION,
phase_frequency_flowervillage_pv DOUBLE PRECISION,
phase_frequency_happyville_pv DOUBLE PRECISION,
phase_frequency_eastwood_pv DOUBLE PRECISION,
phase_frequency_sunrisebay_pv DOUBLE PRECISION,
phase_frequency_brightcounty_pv DOUBLE PRECISION,
a_phase_voltage_cloudington_pv DOUBLE PRECISION,
a_phase_voltage_flowervillage_pv DOUBLE PRECISION,
a_phase_voltage_happyville_pv DOUBLE PRECISION,
a_phase_voltage_eastwood_pv DOUBLE PRECISION,
a_phase_voltage_sunrisebay_pv DOUBLE PRECISION,
a_phase_voltage_brightcounty_pv DOUBLE PRECISION,
b_phase_voltage_cloudington_pv DOUBLE PRECISION,
b_phase_voltage_flowervillage_pv DOUBLE PRECISION,
b_phase_voltage_happyville_pv DOUBLE PRECISION,
b_phase_voltage_eastwood_pv DOUBLE PRECISION,
b_phase_voltage_sunrisebay_pv DOUBLE PRECISION,
b_phase_voltage_brightcounty_pv DOUBLE PRECISION,
c_phase_voltage_cloudington_pv DOUBLE PRECISION,
c_phase_voltage_flowervillage_pv DOUBLE PRECISION,
c_phase_voltage_happyville_pv DOUBLE PRECISION,
c_phase_voltage_eastwood_pv DOUBLE PRECISION,
c_phase_voltage_sunrisebay_pv DOUBLE PRECISION,
c_phase_voltage_brightcounty_pv DOUBLE PRECISION);
```

**Comment by seut** [comment]
@proddata Thank you for that information, we are now able to reproduce the issue and working on a fix.

**Comment by seut** [comment]
@proddata We've fixed this by #10882, it will be part of the next 4.3 release.
Thank you again for reporting.
